articles/why-and-how-zk-snark-works/4.general_purpose_zero_knowledge_proofs.md

```diff
@@ -0,0 +1,146 @@
+---
+Title: General purpose Zero knowledge proofs
+Status: Done
+Level: 3
+---
+# Chapter 4
+## 4.1. Computation
+This step is to translate the program into the polynomial form.
+Example , we have a simple program : 
+```javascript
+function calc(w,a,b)
+	if w then 
+		return a x b
+	else 
+		return a + b
+	else if 
+end function
+```
+
+assuming $\omega \in \{0,1\}$  : 
+$f(\omega,a,b) = \omega (a * b ) + (1- \omega )(a*b)$  
+## 4.2. Single Operation 
+Any computation at it is core consists of elemental operations of the form:
+                 *left operand*  **operator** *right operand* = *output*
+### 4.2.1. Arithmetic Properties of Polynomials
+Let us consider two following polynomials : 
+$f(x) = 2x^{2}-9x+10$
```

Comment:
<img width="700" alt="Screenshot 2024-01-09 at 00 17 20" src="https://github.com/sota-zk-lab/zkp-documents/assets/33181397/28cde312-4a39-4f1a-aa7d-cbdfe3490d6c">
Fix the format